electron/src/services/electronCrypto.service.ts

```diff
@@ -1,69 +1,74 @@
-        }
+  protected async storeKey(key: SymmetricCryptoKey, userId?: string) {
+    if (await this.shouldStoreKey(KeySuffixOptions.Auto, userId)) {
+      await this.stateService.setCryptoMasterKeyAuto(key.keyB64, { userId: userId });
+    } else {
+      this.clearStoredKey(KeySuffixOptions.Auto);
     }
 
-    protected async retrieveKeyFromStorage(keySuffix: KeySuffixOptions) {
-        await this.upgradeSecurelyStoredKey();
-        return super.retrieveKeyFromStorage(keySuffix);
+    if (await this.shouldStoreKey(KeySuffixOptions.Biometric, userId)) {
+      await this.stateService.setCryptoMasterKeyBiometric(key.keyB64, { userId: userId });
+    } else {
+      this.clearStoredKey(KeySuffixOptions.Biometric);
     }
+  }
 
-    /**
-     * @deprecated 4 Jun 2021 This is temporary upgrade method to move from a single shared stored key to
-     * multiple, unique stored keys for each use, e.g. never logout vs. biometric authentication.
-     */
-    private async upgradeSecurelyStoredKey() {
-        // attempt key upgrade, but if we fail just delete it. Keys will be stored property upon unlock anyway.
-        const key = await this.stateService.getCryptoMasterKeyB64();
+  protected async retrieveKeyFromStorage(keySuffix: KeySuffixOptions) {
+    await this.upgradeSecurelyStoredKey();
+    return super.retrieveKeyFromStorage(keySuffix);
+  }
 
-        if (key == null) {
-            return;
-        }
+  /**
+   * @deprecated 4 Jun 2021 This is temporary upgrade method to move from a single shared stored key to
+   * multiple, unique stored keys for each use, e.g. never logout vs. biometric authentication.
+   */
+  private async upgradeSecurelyStoredKey() {
+    // attempt key upgrade, but if we fail just delete it. Keys will be stored property upon unlock anyway.
+    const key = await this.stateService.getCryptoMasterKeyB64();
 
-        try {
-            if (await this.shouldStoreKey(KeySuffixOptions.Auto)) {
-                await this.stateService.setCryptoMasterKeyAuto(key);
-            }
-            if (await this.shouldStoreKey(KeySuffixOptions.Biometric)) {
-                await this.stateService.setCryptoMasterKeyBiometric(key);
-            }
-        } catch (e) {
-            this.logService.error(`Encountered error while upgrading obsolete Bitwarden secure storage item:`);
-            this.logService.error(e);
-        }
+    if (key == null) {
+      return;
+    }
 
-        await this.stateService.setCryptoMasterKeyB64(null);
+    try {
+      if (await this.shouldStoreKey(KeySuffixOptions.Auto)) {
+        await this.stateService.setCryptoMasterKeyAuto(key);
+      }
+      if (await this.shouldStoreKey(KeySuffixOptions.Biometric)) {
+        await this.stateService.setCryptoMasterKeyBiometric(key);
+      }
+    } catch (e) {
+      this.logService.error(
+        `Encountered error while upgrading obsolete Bitwarden secure storage item:`
+      );
+      this.logService.error(e);
```

Comment:
No need for this to be a template string, so should be double-quote. Or include the e in the string, then it can stay a template string

Reply:
True, but we should avoid doing any manual intervention at this stage, since all existing branches will have to sync against this using the above method which would throw away the manual operations.